app/models/course/assessment.rb

```diff
@@ -171,6 +171,15 @@ def include_in_consolidated_email?(event)
                                                           "assessment_#{event}".to_sym)
   end
 
+  def graded_test_case_types
+    test_case_types = Course::Assessment::Question::ProgrammingTestCase.test_case_types
+    graded_test_case_types = []
```

Comment:
I think you can write this with `tap`. See https://github.com/Coursemology/coursemology2/blob/bf2638c88f5c57ec939d82855d9b24c4058e009f/app/services/course/assessment/submission/update_service.rb#L85